### ISSUE TITLE
Any pipe (|> or %>%) on the end of a line leads to a false negative for commented_code_linter

### ISSUE THREAD

**Issue by jcken95** [r-lib/lintr#2671]
``` r
##  no pipe

lintr::lint(
  text = "# mean()"
)
#> <text>:1:3: style: [commented_code_linter] Remove commented code.
#> # mean()
#>   ^~~~~~

## base pipes

lintr::lint(
  text = "# mean() |>"
)
#> ℹ No lints found.

lintr::lint(
  text = "# 1:5 |> mean()"
)
#> <text>:1:3: style: [commented_code_linter] Remove commented code.
#> # 1:5 |> mean()
#>   ^~~~~~~~~~~~~

## magrittr pipes

lintr::lint(
  text = "# mean() %>%"
)
#> ℹ No lints found.

lintr::lint(
  text = "# 1:5 %>% mean()"
)
#> <text>:1:3: style: [commented_code_linter] Remove commented code.
#> # 1:5 %>% mean()
#>   ^~~~~~~~~~~~~~
```

<sup>Created on 2024-10-18 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

----

``` r
# .lintr

linters: linters_with_defaults(
           line_length_linter(120L),
           object_length_linter(40L),
           pipe_consistency_linter("|>"),
           object_usage_linter = NULL,
           indentation_linter = NULL,
           unreachable_code_linter()
         )
encoding: "UTF-8"

```

**Comment by MichaelChirico** [comment]
Sorry, what's the expected behavior here?

In order for code to be detected, it has to parse. Your examples with no lints don't parse.

**Comment by jcken95** [comment]
The expected behaviour would be for `lintr::lint(text = " # mean() |>"` to return a commented code linter.

Appreciate this code is not parsable in isolation, however, code of this form is a common occurrence in data wrangling. For example:

``` r
# no lint detected (expected, desireable behaviour)
lintr::lint(
  text = "mtcars |>
           #dplyr::filter(cyl > 4) |>
            dplyr::summarise(m = mean(wt))",
  linters = lintr:::commented_code_linter()
)

# would like/expect the commented out `dplyr::filter()` to picked up by the lintr
# but commented_code_linter() is not flagged in this example, because the single line
# is not parsable
lintr::lint(
  text = "mtcars |>
           #dplyr::filter(cyl > 4) |>
            dplyr::summarise(m = mean(wt))",
  linters = lintr:::commented_code_linter()
)
```

<sup>Created on 2024-10-19 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>
